langchain_mcp_adapters/client.py

```diff
@@ -321,11 +321,15 @@ async def connect_to_server_via_websocket(
 
         await self._initialize_session_and_load_tools(server_name, session)
 
-    def get_tools(self) -> list[BaseTool]:
+    def get_tools(self, server_name: str | None = None) -> list[BaseTool]:
         """Get a list of all tools from all connected servers."""
         all_tools: list[BaseTool] = []
-        for server_tools in self.server_name_to_tools.values():
+        if server_name:
+            server_tools = self.server_name_to_tools.get(server_name, [])
```

Comment:
you can just return `server_tools` directly here, no need for adding it to `all_tools`, eg.

```python
def get_tools(self, server_name: str | None = None) -> list[BaseTool]:
    if server_name:
        if server_name not in self.server_name_to_tools:
            raise ValueError(f"Couldn't find a server with name '{server_name}', expected one of '{list(self.server_name_to_tools.keys())}'")
        return self.server_name_to_tools[server_name]

    all_tools: list[BaseTool] = []
    for server_tools in self.server_name_to_tools.values():
        all_tools.extend(server_tools)

    return all_tools
```